### PR TITLE
Update text and default for announcement notification; fix user edit form

### LINKED ISSUE #5600: Update text for developer notification checkbox "add-on events and contests"
Currently, developers see these notification options when they edit their account settings on https://addons.mozilla.org/en-US/firefox/users/edit: 

notifications

Following a discussion with @amyt and @jvillalobos, we would like to update the text next to the box  "add-on contests or events are announced" to "stay up to date with news and events relevant to add-on developers." 

Whatever parameter is being passed in the backend can stay the same; we just want to change the text associated with that box. 

Users who opt in to this box will be added to the subscription list for the about:addons newsletter. 

Thank you!

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #5600

### CODE REVIEW

File: src/olympia/users/forms.py [hunk 1]
```diff
@@ -210,7 +210,11 @@ def save(self, log_for_developer=True):
             tasks.resize_photo.delay(tmp_destination, u.picture_path,
                                      set_modified_on=[u])
 
-        for (i, n) in notifications.NOTIFICATIONS_BY_ID.items():
+        visible_notifications = (
+            notifications.NOTIFICATIONS_BY_ID if self.instance.is_developer
+            else notifications.NOTIFICATIONS_BY_ID_NOT_DEV)
+
+        for (i, n) in visible_notifications.items():
```
Comment (EnTeQuAk) [inline src/olympia/users/forms.py]: I know that this is old code but can we rename `n` and `i` to something more proper lilke `notification` and `notification_id`?

File: src/olympia/users/notifications.py [hunk 2]
```diff
@@ -66,9 +66,10 @@ class announcements(_NOTIFICATION):
     id = 8
     group = 'dev'
     short = 'announcements'
-    label = _("add-on contests or events are announced")
+    label = _("stay up-to-date with news and events relevant to add-on "
+              "developers")
```
Comment (EnTeQuAk) [inline src/olympia/users/notifications.py]: this should totally use single quotes

Comment (eviljeff) [inline src/olympia/users/notifications.py]: All of the rest of the file used doubles for the text so it was consistent at least.  But I'll change (this one).

**Comment by eviljeff** [comment]
I'd replace all of this crappy old code if the user profile pages weren't being replaced with a front-end rewrite in the next few months.

File: src/olympia/users/notifications.py (hunk 2 above)
Comment (EnTeQuAk) [inline src/olympia/users/notifications.py]: fair point, then i think consistency wins…

Comment (eviljeff) [inline src/olympia/users/notifications.py]: 🙄 I just changed them all to single quotes

Comment (eviljeff) [inline src/olympia/users/notifications.py]: https://github.com/mozilla/addons-server/pull/5622/commits/e0e3ec740abaf69570ac2a86b3564e7ae82412e4

File: src/olympia/users/forms.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/users/forms.py]: https://github.com/mozilla/addons-server/pull/5622/commits/e0e3ec740abaf69570ac2a86b3564e7ae82412e4

**Pull request merged**